signer/src/transaction_signer.rs

```diff
@@ -87,8 +87,10 @@ pub struct TxSignerEventLoop<Network, Storage, BlocklistChecker> {
     pub block_observer_notifications: tokio::sync::watch::Receiver<()>,
     /// Private key of the signer for network communication.
     pub signer_private_key: p256k1::scalar::Scalar,
-    /// How many blocks back from the chain tip the signer will look for requests.
-    pub context_window: usize,
+    /// How many bitcoin blocks back from the chain tip the signer will look for deposit requests.
+    pub bitcoin_context_window: usize,
+    /// How many stacks blocks back from the chain tip the signer will look for withdraw requests.
+    pub stacks_context_window: usize,
```

Comment:
Shouldn't the stacks context window reference bitcoin blocks? I would think that any of the signers' guarantees are tied to bitcoin blocks due to the fact that they need to look sufficiently far back within coordinator changes, and those are measured by bitcoin blocks.

I'm thinking the "guarantee" is something like "we will look back within 3 coordinator changes for any withdrawal requests". Later we can make the guarantee stronger by looking at distinct coordinators, which will be a better guard against failed coordinators.

Reply:
Yeah, I mean we discussed cutting the cord between bitcoin and stacks so I did this because it was 1. easier to implement and 2. relies less on the connection between bitcoin and stacks. But since it has turned out to be simpler to find which stacks blocks are confirmed by bitcoin blocks than we previously anticipated, I'm all in favor of relying more extensively on this connection.

I'll update and get rid of the stacks context window.

Reply:
Done as commented in the other thread. I'll do some additional code clean up as well.